PEP 561 test failures due to pyinfo break wheel builds on macOS

Traceback from https://github.com/mypyc/mypy_mypyc-wheels/runs/7376619798?check_suite_focus=true (also reproduces locally on macOS):
```
  ______________________________ testTypedPkgSimple ______________________________
  [gw1] darwin -- Python 3.10.0 /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmpl3v_i2pp/bin/python
  data: /Users/runner/work/mypy_mypyc-wheels/mypy_mypyc-wheels/mypy/test-data/unit/pep561.test:14:
  ../tmpl3v_i2pp/lib/python3.10/site-packages/mypy/test/testpep561.py:31: in run_case
      test_pep561(test_case)
  ../tmpl3v_i2pp/lib/python3.10/site-packages/mypy/test/testpep561.py:142: in test_pep561
      out, err, returncode = mypy.api.run(cmd_line)
  mypy/api.py:69: in run
      ???
  mypy/api.py:58: in _run
      ???
  mypy/api.py:69: in <lambda>
      ???
  mypy/main.py:96: in main
      ???
  mypy/main.py:173: in run_build
      ???
  mypy/build.py:154: in build
      ???
  mypy/build.py:186: in _build
      ???
  mypy/modulefinder.py:844: in compute_search_paths
      ???
  mypy/modulefinder.py:748: in get_search_dirs
      ???
  /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/subprocess.py:420: in check_output
      return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/subprocess.py:524: in run
      raise CalledProcessError(retcode, process.args,
  E   subprocess.CalledProcessError: Command '['/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmp1l0jvq71/bin/python', '/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmpl3v_i2pp/lib/python3.10/site-packages/mypy/pyinfo.py', 'getsearchdirs']' returned non-zero exit status 1.
```

I have checked that this is a side effect of #13161.

--- a/mypy/pyinfo.py
+++ b/mypy/pyinfo.py
@@ -14,6 +14,15 @@ MYPY = False
 if MYPY:
     from typing import List
 
+if __name__ == '__main__':
+    # HACK: We don't want to pick up mypy.types as the top-level types
+    #       module. This could happen if this file is run as a script.
+    #       This workaround fixes it.
+    old_sys_path = sys.path
+    sys.path = sys.path[1:]
+    import types  # noqa
+    sys.path = old_sys_path
+
 
 def getsearchdirs():
     # type: () -> List[str]

